ajson: add page (#4191)
tldr-pages/tldr: pages/common/ajson.md

# ajson

> > Executes JSONPath with JSON document.

Reviewer: Perhaps this will read better?
```suggestion
> Executes JSONPath on JSON objects.
```

> More information: <https://github.com/spyzhov/ajson>.

> - Read JSON from a file and execute JSONPath:

Reviewer: Perhaps this will read better?
```suggestion
- Read JSON from a file and execute a specified JSONPath expression:
```

> `ajson "$..json[?(@.path)]" example.json`

Reviewer: I'd probably advise single quotes here instead of double quotes to avoid issues with those special characters. Also, perhaps we could use the `{{token_syntax}}` here and make `example.json` more consistent?

```suggestion
`ajson '{{$..json[?(@.path)]}}' {{path/to/file.json}}`
```

> - Read JSON from STDIN and execute JSONPath:

Reviewer: Perhaps this will read better?
```suggestion
- Read JSON from stdin and execute a specified JSONPath expression:
```

> `cat example.json | ajson "$..json[?(@.path)]"`

Reviewer: Same here with the filename and quotes.
```suggestion
`cat {{path/to/file.json}} | ajson '{{$..json[?(@.path)]}}'`
```

> - Read JSON from `GET` request and calculate value:

Reviewer: Perhaps this will read better?
```suggestion
- Read JSON from a URL and evaluate a specified JSONPath expression:
```

> `ajson "avg($..price)" "https://example.com/api/"`

Reviewer: Perhaps this will read better?
```suggestion
`ajson '{{avg($..price)}}' "{{https://example.com/api/}}"`
```

> - Curl JSON from `POST` request and calculate value:

Reviewer: This example seems to be demonstrating the same functionality of `ajson` as the previous stdin example. Generally in tldr pages we try to focus on the functionality of the command itself and not try to show examples of other commands. Perhaps this example could be removed?

`curl -s -d "param=value" -X POST http://example.com/ | ajson "sum($..price)"`

> - Read simple JSON and calculate value:

Reviewer: Perhaps this will read better?
```suggestion
- Read some simple JSON and calculate a value:
```

> `echo "3" | ajson "2 * pi * $"`

Reviewer: Same here again with the quotes &  `{{token_syntax}}`.
```suggestion
`echo "{{3}}" | ajson '{{2 * pi * $}}'`
```
